lib/mongo/server/monitor/connection.rb

```diff
@@ -240,6 +244,14 @@ def add_server_connection_id
           end
           raise e
         end
+
+        # Detect whether server supports hello command based on server reply
```

Comment:
This method updates `@hello_ok` based on server support of hello. I would expect a method that only detects something to not mutate the receiver.

Reply:
Updated the docsctring.